### PR TITLE
Add apps to restriction rules in branch protection

### PR DESCRIPTION
**Pull request by jporzucek**
Fixes https://github.com/google/go-github/issues/2506

### CODE REVIEW

File: go.mod
Comment (gmlewis) [inline go.mod:12]: Please revert all changes to `go.mod` and `go.sum` as they are unrelated to this PR.

File: go.sum
Comment (gmlewis) [inline go.sum:10]: Please revert all changes to `go.mod` and `go.sum` as they are unrelated to this PR.

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @jporzucek .

**Review by gmlewis — APPROVED** [review]
Thank you, @jporzucek !
LGTM.

Awaiting second LGTM+Approval from any other contributor to this repo before merging.

**Review by raynigon — APPROVED** [review]
LGTM

**Pull request merged**
